tidy3d/components/simulation.py

```diff
@@ -2217,6 +2359,65 @@ def _get_structures_plane(
                     medium_shapes.append((structure.medium, shape))
         return medium_shapes
 
+    @staticmethod
+    def _get_structures_2dbox(
+        structures: List[Structure],
+        x: float = None,
+        y: float = None,
+        z: float = None,
+        hlim: Tuple[float, float] = None,
+        vlim: Tuple[float, float] = None,
+    ) -> List[Tuple[Medium, Shapely]]:
+        """Compute list of shapes to plot on 2d box specified by (x_min, x_max), (y_min, y_max).
+
+        Parameters
+        ----------
+        structures : List[:class:`.Structure`]
+            list of structures to filter on the plane.
+        x : float = None
+            position of plane in x direction, only one of x, y, z must be specified to define plane.
+        y : float = None
+            position of plane in y direction, only one of x, y, z must be specified to define plane.
+        z : float = None
+            position of plane in z direction, only one of x, y, z must be specified to define plane.
+        hlim : Tuple[float,float] = None
+            Horizontal bounds for plotting
+        vlim : Tuple[float,float] = None
+            Vertical bounds for plotting
+
+        Returns
+        -------
+        List[Tuple[:class:`.AbstractMedium`, shapely.geometry.base.BaseGeometry]]
+            List of shapes and mediums on the plane.
+        """
+        (h_min, h_max), (v_min, v_max) = hlim, vlim
+        h_length, v_length = h_max - h_min, v_max - v_min
+        if z == None:
```

Comment:
there exists a `self.unpop_axis()` [docs](https://docs.flexcompute.com/projects/tidy3d/en/latest/_autosummary/tidy3d.Sphere.html?highlight=unpop%20axis#tidy3d.Sphere.unpop_axis) that could be useful here to simplify things. Here's one suggestion

```py
(h_min, h_max), (v_min, v_max) = hlim, vlim

# get center and size in the h, v "basis"
h_center = (h_min + h_max) / 2.0
v_center = (v_min + v_max) / 2.0
h_size = h_max + h_min 
v_size = v_max + v_min

axis, center_normal = self.parse_xyz_kwargs(x=x, y=y, z=z)
center = self.unpop_axis(center_normal, (h_center, v_center), axis=axis)
size = self.unpop_axis(0.0, (h_size, v_size), axis=axis)

# continue with line 2410
```

unpop axis is the inverse of pop axis and I think handles conversion from h, v to x, y, z more elegantly.

Reply:
h_size = h_max - h_min but yes otherwise this works elegantly. changed.